python/lsst/obs/base/ingest.py

```diff
@@ -259,65 +317,71 @@ def extractMetadata(self, filename: str) -> RawFileData:
             A structure containing the metadata extracted from the file,
             as well as the original filename.  All fields will be populated,
             but the `RawFileData.dataId` attribute will be a minimal
-            (unexpanded) `DataCoordinate` instance.
+            (unexpanded) `~lsst.daf.butler.DataCoordinate` instance. The
+            ``instrumentClsas`` field will be `None` if there is a problem
+            with metadata extraction.
 
         Notes
         -----
         Assumes that there is a single dataset associated with the given
         file.  Instruments using a single file to store multiple datasets
         must implement their own version of this method.
-        """
 
-        # We do not want to stop ingest if we are given a bad file.
-        # Instead return a RawFileData with no datasets and allow
-        # the caller to report the failure.
+        By default the method will catch all exceptions unless the ``failFast``
+        configuration item is `True`.  If an error is encountered the
+        `_on_metadata_failure()` method will be called. If no exceptions
+        result and an error was encountered the returned object will have
+        a null-instrument class and no datasets.
 
+        This method supports sidecar JSON files which can be used to
+        extract metadata without having to read the data file itself.
+        The sidecar file is always used if found.
+        """
+        sidecar_fail_msg = ""
```

Comment:
I'd note here that this message needs a prepended space to get the log formatting to work right.